src/deepsparse/v2/pipeline.py

```diff
@@ -93,59 +98,84 @@ def _apply_split(self, inp: Any, inference_state: InferenceState):
         """
 
         batches, orig_batch_size = self.expand_inputs(inp, 1)
-        run_with_state = partial(
-            self._run_sequential,
-            pipeline_state=self.pipeline_state,
-            start=self.router.route[self.router.SPLIT_ROUTE],
-            end=self.router.JOIN_ROUTE,
-        )
-        inference_state_list = [
-            copy.deepcopy(inference_state) for x in range(len(batches))
+
+        # Create a list of SplitRoutes, per batch size 1
+        # Each SplitRoute object holds information about the particular path it
+        # follows. All start at the same step defined by SPLIT_ROUTE and start
+        # with the same inference_state.
+        split_routes = [
+            SplitRoute(
+                inf=copy.deepcopy(inference_state),
+                step=self.router.route[self.router.SPLIT_ROUTE],
+            )
+            for i in range(len(batches))
         ]
-        futures = self._scheduler_group.map(
-            batches,
-            inference_state_list,
-            func=run_with_state,
-        )
-        return self.condense_inputs([x.result() for x in futures])
 
-    def _run_next_step(
+        # Start each SPLIT_ROUTE; store the Future for each.
+        for i in range(len(split_routes)):
```

Comment:
would be good to move this into a separate helper function/file that takes in the scheduler, subgraphs, etc and performs the scheduling loop